AnkiHubSoftware/ankihub_addon
# fix: Improve AnkiHub DB check

This PR makes some changes to the AnkiHub DB check to raise errors which are currently silently ignored and to remove data for not-existing decks from the add-on database.

## Proposed changes
- Raise errors which happen during deck installations (the result of background operations is currently not checked)
-  Uninstall decks that are not on AnkiHub anymore (because they were deleted on the web app)

## How to reproduce
- Subscribe to a deck
- Remove the deck from AnkiHub 
- Remove the deck entry from the private config
- Sync with AnkiHub
- The dialog from the AnkiHub database check will open
- Choose yes
- The deck was uninstalled silently
- You can restart Anki and sync with AnkiHub, the dialog from the AnkiHub database check won't open anymore because the deck was removed from the database during the uninstallation

<a href="https://gitpod.io/#https://github.com/ankipalace/ankihub_addon/pull/758"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>
